resources/table/table.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+
+interface TableProps<T> {
+  data: T[];
+  columns: ColumnConfig<T>[];
+}
+
+type Direction = "asc" | "desc";
+
+const Table = <T,>({ data, columns }: TableProps<T>) => {
+  const [sortedData, setSortedData] = useState(data);
+  const [sortConfig, setSortConfig] = useState<{
+    key: keyof T;
+    direction: Direction;
+  } | null>(null);
+
+  const handleSort = (column: ColumnConfig<T>) => {
+    if (!column.sorter || !column.dataIndex) return;
+
+    let direction: Direction = "asc";
+    if (
+      sortConfig &&
+      sortConfig.key === column.dataIndex &&
+      sortConfig.direction === "asc"
+    ) {
+      direction = "desc";
+    }
+
+    const sorted = [...data].sort((a, b) => {
+      if (a[column.dataIndex!] < b[column.dataIndex!])
+        return direction === "asc" ? -1 : 1;
+      if (a[column.dataIndex!] > b[column.dataIndex!])
+        return direction === "asc" ? 1 : -1;
+      return 0;
+    });
+
+    setSortConfig({ key: column.dataIndex, direction });
+    setSortedData(sorted);
+  };
```

Comment:
Sorted may be handle by the API.
Maybe we need a flag, if is not handled by api we can paginate and sort in local.
If it is, then we don't do it.

Reply:
Yeah, in the lazyload senario we should sort via API, from here we will just emit a event on sort change.